src/main/scala/impl/group5/vrefAnalysis.scala

```diff
@@ -0,0 +1,264 @@
+
+        //Überprüfung auf load und store instruktion
+        method.body match {
+          case None =>
+          case Some(code) =>code.instructions.foreach {
+            case load: LoadLocalVariableInstruction =>
+              if(loadInstructionmap.contains(load))
+              {
+
+                loadInstructionmap(load) = loadInstructionmap(load)+1
+              }
+              else
+              {
+                loadInstructionmap += (load -> 1)
+              }
+
+            case store: StoreLocalVariableInstruction =>
+
+                if(storeInstructionmap.contains(store))
+                {
+
+                  storeInstructionmap(store) = storeInstructionmap(store)+1
+                }
+                else
+                {
+                  storeInstructionmap += (store -> 1)
+                }
+
+            case nullInstruction: LoadConstantInstruction[Null] => if(nullInstruction.opcode ==1) nullReferenz += 1
+
+            case putfield: PUTFIELD =>
+              if(putfieldCount.contains(putfield))
+              {
+
+                putfieldCount(putfield) = putfieldCount(putfield)+1
+              }
+              else
+              {
+                putfieldCount += (putfield -> 1)
+              }
+
+            case getfield: GETFIELD =>
+              if(getfieldCount.contains(getfield))
+              {
+
+                getfieldCount(getfield) = getfieldCount(getfield)+1
+              }
+              else
+              {
+                getfieldCount += (getfield -> 1)
+              }
+            case _ =>
+          }
+        }
+
+
+        var sumLoad = 0
+        var sumStore = 0
+        var sumthisLoad = 0
+        var sumthisStore = 0
+        var sumFieldLoad = 0
+        var sumFieldStore = 0
+
+        //Summiert einzelne load und store Referenzen auf
+        loadInstructionmap.foreach( y => sumLoad += y._2)
+        storeInstructionmap.foreach(y => sumStore += y._2)
+        getfieldCount.foreach(y => sumFieldLoad += y._2)
+        putfieldCount.foreach(y => sumFieldStore += y._2)
+
+
+        //Guckt auf this Referenzen und zieht die von den aufsummerten  load und store Referenzen
+        if(localtable.nonEmpty)
+        {
+
+          var thisIndex = -1
+          val  list=   localtable.get
+          list.foreach(y =>
+            if(y.name =="this") {
+              thisIndex = y.index
+            })
+          loadInstructionmap.foreach(y => {
+              if(y._1.lvIndex==thisIndex) sumthisLoad = y._2
+          })
+          storeInstructionmap.foreach(y => {
+            if(y._1.lvIndex==thisIndex) sumthisLoad = y._2
+          })
+          sumLoad -= sumthisLoad
+          sumStore -= sumthisStore
+        }
+        else if(sumFieldLoad !=0 || sumFieldStore !=0 || sumLoad !=0 || sumStore !=0) {
+          log.warn("this kann nicht aufgelösst werden und wird als Referenz mitgezählt, da die jar die localVariableTable nicht mitliefert")
```

Comment:
Etwas störend:
Das es so oft keine localVariableTable gibt macht es das log sehr unübersichtlich.

Reply:
Nun die localVariableTable wird nicht immer mitgeliefert und das kann ich nicht ändern.

Reply:
Ja das habe ich auch gemerkt.  Fand dadurch nur das Log unübersichtlich da quasi 90 % des logs die folgende Nachricht enthält:
"this kann nicht aufgelösst werden und wird als Referenz mitgezählt, da die jar die localVariableTable nicht mitliefert".

Das könnte man vielleicht irgendwie reduzieren, ein Counter der das hochzählt und am Ende insgesamt ausgibt oder so. 
War auch das einzige was mich irritiert hat und von der Logik funktioniert ja alles.